src/charm.py

```diff
@@ -481,134 +43,61 @@ def __init__(self, *args) -> None:  # type: ignore[no-untyped-def]
-
-    def _on_start(self, _: StartEvent) -> None:
-        """Handle the start event."""
-        # We need to set ActiveStatus here because this is a workload-less
-        # charm, so there's no pebble-ready event to react to. This means this
-        # is the only event (outside of update-status) that is fired if a pod is
-        # restarted by the k8s cluster. If we don't do anything here the charm
-        # would remain in maintenance status.
-        self.unit.status = ActiveStatus()
-
-    @staticmethod
-    def _gen_relation_dedup_key(relation: Relation) -> typing.Tuple[str, ...]:
-        """Generate a key from the given relation to detect duplicates in _deduped_relations.
-
-        Args:
-            relation: the given relation object.
-
-        Returns: The key as a tuple of strings.
-
-        Raises:
-            RuntimeError: if the remote application is unknown.
-        """
-        app = relation.app
-        if not app:
-            raise RuntimeError(f"can't retrieve remote application from relation {relation}")
-        data = relation.data[app]
-        return (
-            data.get("service-hostname"),
-            data.get("service-name"),
-            data.get("service-model"),
-            data.get("service-port"),
-            app.name,
-        )
-
-    def _deduped_relations(self) -> Tuple[ops.Relation, ...]:
-        """Return a relation list with duplicates removed.
-
-        Relations are considered duplicated if they meet the following criteria:
-        1. Two are connected via the legacy ingress and the nginx-route relation endpoint
-        2. Two share the same 4-tuple (service-hostname, service-name, service-model, service-port)
-        3. Both relations have the same remote application.
-
-        In the case of duplicates, the relation connected via legacy ingress is removed.
-
-        Returns:
-            A relation list with duplicates removed.
-        """
-        nginx_route_relations = [
-            r for r in self.model.relations["nginx-route"] if r.app is not None
-        ]
-        ingress_relations = [r for r in self.model.relations["ingress"] if r.app is not None]
-        nginx_route_relation_keys = set(
-            self._gen_relation_dedup_key(r) for r in nginx_route_relations
-        )
-        dedup_ingress_relations = []
-        for relation in ingress_relations:
-            if self._gen_relation_dedup_key(relation) in nginx_route_relation_keys:
-                LOGGER.warning(
-                    "legacy ingress relation from app %s is shadowed by nginx-route relation",
-                    relation.app,
-                )
-                continue
-            dedup_ingress_relations.append(relation)
-        return tuple(nginx_route_relations + dedup_ingress_relations)
-
-    @property
-    def _all_config_or_relations(self) -> Any:
-        """Get all configuration and relation data.
+    def _get_ingress_options(self) -> Optional[IngressOption]:
```

Comment:
This should not return None, it should just return `IngressOptions`

Reply:
Perhaps separate the logic for whether the relation is present/ whether cleanup is needed